src/components/gocam-legend/gocam-legend.tsx

```diff
@@ -42,26 +13,59 @@ export class GocamLegend {
   render() {
     return (
       <Host>
-        <div class='header' part="header">Relation Types</div>
-        <div class='sections'>
-          {Object.keys(legend).map((section) => {
-            return (
-              <div class={'section ' + section} part="section">
-                {legend[section].map((item) => {
-                  return (
-                    <div class='item'>
-                      <img alt={item.label} src={IMAGE_DATA[item.id]}></img>
-                      <div class='item-label'>
-                        {item.label}
-                      </div>
-                    </div>
-                  );
-                })}
-              </div>
-            );
-          })}
+        <svg height="0" width="0" style={{ position: 'absolute' }}>
+          <defs>
+            {Object.entries(RELATION_MAP).map(([id, config]) => (
+              config.glyph === 'circle-triangle' ?
+                <marker id={`${config.glyph}-${id}`}
+                  viewBox="0 0 20 10"
+                  refX="18"
+                  refY="5"
+                  markerWidth="12"
+                  markerHeight="4"
+                  orient="auto">
+                  <circle cx="7" cy="5" r="3.5" fill={config.color} />
+                  <path d="M 12 0 L 18 5 L 12 10 z" fill={config.color} />
+                </marker>
+                :
+                <marker id={`${config.glyph}-${id}`} viewBox="0 0 10 10" refX="9" refY="5"
+                  markerWidth="4" markerHeight="4" orient="auto">
+                  {config.glyph === 'triangle' && <path d="M 0 0 L 10 5 L 0 10 z" fill={config.color} />}
+                  {config.glyph === 'tee' && <rect x="6" y="0" width="4" height="10" fill={config.color} />}
+                  {config.glyph === 'circle' && <circle cx="5" cy="5" r="4" fill={config.color} />}
+                  {config.glyph === 'square' && <rect x="0" y="0" width="10" height="10" fill={config.color} />}
+                </marker>
+            ))}
+          </defs>
+        </svg>
+
+        <div class='header'>Relation Types</div>
+        <div class="columns">
+          {Object.entries(LEGEND_COLUMNS).map(([columnName, relations]) => (
+            <div class={`column ${columnName}`}>
```

Comment:
so to to make it like before I made sections and section. This is more scalable in case a client wanna make them rows. Any thoughts?

``` 
/**
 * @part header - The header
 * @part sections - Group of legend entries
 * @part section - An individual legend entry
 */
```

and also will update the parent component

Reply:
Makes sense to me